proposals/20230323-end-to-end-testing.md

```diff
@@ -0,0 +1,372 @@
+# End-to-end Testing
+
+**Authors**: Dominik Fleischmann ([@domFleischmann](https://github.com/domFleischmann)),
+Kimonas Sotirchos ([@kimwnasptd](https://github.com/kimwnasptd)),
+and Anna Jung ([@annajung](https://github.com/annajung))
+
+## Background
+
+Previously, the Kubeflow community leveraged prow optional-test-infra for e2e testing with credit from AWS. After the
+optional test infrastructure deprecation notice, all WGs moved their test to GitHub Actions as a temporary solution. Due
+to resource constraints of GitHub-hosted runners, the Kubeflow community stopped supporting e2e tests as part of the
+migration. In partnership with Amazon, a new AWS account has been created with sponsored credits. With the new AWS
+account, the Kubeflow community is no longer limited by resource constraints posed by GitHub Actions. To enable the e2e
+test for the Manifest repo, this doc proposes a design to set up the infrastructure needed to run the necessary tests.
+
+References
+
+- [Optional Test Infra Deprecation Notice](https://github.com/kubeflow/testing/issues/993)
+- [Alternative solution to removal of test on optional-test-infra](https://github.com/kubeflow/testing/issues/1006)
+
+## Goal
+
+Enable the e2e testing for the Manifest repo and leverage it to shorten the manifest testing phase of the Kubeflow
+release cycle and to increase quality of the Kubeflow release by ensuring Kubeflow components and dependencies work
+correctly together.
+
+## Proposal
+
+After some initial conversations, it has been agreed to create integration tests based on GitHub Actions, which will
+spawn an EC2 instance with enough resources to deploy the complete Kubeflow solution and run some end-to-end testing.
+
+### Alternative solutions considered
+
+#### Prow
+
+While there are some existing tests with Prow, those tests were discarded due to them not having been updated in 2 years
+and there being a high amount of complexity in these tests. After some investigation, the Manifests Working Group
+decided that it would be more work adapting those tests to the current state of manifests than starting from scratch
+with lower complexity.
+
+#### Self-hosted runners
+
+Self-hosted runners are not recommended with public repositories due to security concerns with how it behaves on a pull
+request made by a forked repository.
+
+## Implementation
+
+Below lists steps the GitHub actions will perform to complete end-to-end testing
+
+- [Create an EC2 instance](#create-an-ec2-instance)
+- [Install a Kubernetes on the instance](#install-a-kubernetes-on-the-instance)
+- [Deploy Kubeflow](#deploy-kubeflow)
+- [Run tests](#run-tests)
+- [Log and report errors](#log-and-report-errors)
+- [Clean up](#clean-up)
+
+### Create an EC2 instance
+
+Access the AWS credentials (stored as GH Secrets) and create an EC2 instance
```

Comment:
Could expose some more info here regarding what are the type of info that we'll need to be storing as GH Secrets? Like:
1. The AWS account name
2. The key used for authenticating 
3. etc...

Reply:
I created another step in the beginning called "Create credentials required by the AWS" to address what variables need to be created. PTAL

Reply:
Looks great, thanks!